hasher-matcher-actioner/webapp/src/settings/ActionRuleSettingsTab.jsx

```diff
@@ -22,7 +103,155 @@ export default function ActionRuleSettingsTab() {
             </p>
           </Col>
         </Row>
+        <Row>
+          <Col>
+            <Table bordered>
+              <thead>
+                <tr>
+                  <th>
+                    <Button
+                      className="table-action-button"
+                      onClick={() => {
+                        setAdding(true);
+                      }}>
+                      <ion-icon name="add" size="large" />
+                    </Button>
+                  </th>
+                  <th>Name</th>
+                  <th>Labeled As</th>
+                  <th>Not Labeled As</th>
+                  <th>Action</th>
+                </tr>
+              </thead>
+              <tbody>
+                <tr hidden={!adding}>
+                  <td>
+                    <Button
+                      variant="outline-primary"
+                      className="mb-2 table-action-button"
+                      onClick={() => {
+                        const newActionRule = {
+                          name: nameRef.current.value,
+                          must_have_labels: mustHaveLabelsRef.current.value,
+                          must_not_have_labels:
+                            mustNotHaveLabelsRef.current.value,
+                          action_id: actionRef.current.value,
+                        };
+
+                        actionRules.push(newActionRule);
+
+                        actionRules.sort((a, b) =>
+                          a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1,
+                        );
+
+                        setAdding(false);
+                      }}>
+                      <ion-icon
+                        name="checkmark"
+                        size="large"
+                        className="ion-icon-white"
+                      />
+                    </Button>{' '}
```

Comment:
is `{' '}` on purpose?

Reply:
Yes. Prettier does that as one of its standards.